src/components/DecisioningEngine/index.js

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import { noop, sanitizeOrgIdForCookieName } from "../../utils";
+import createOnResponseHandler from "./createOnResponseHandler";
+import createDecisionProvider from "./createDecisionProvider";
+import createApplyResponse from "./createApplyResponse";
+import createEventRegistry from "./createEventRegistry";
+import createContextProvider from "./createContextProvider";
+
+const createDecisioningEngine = ({ config, createNamespacedStorage }) => {
+  const { orgId } = config;
+  const storage = createNamespacedStorage(
+    `${sanitizeOrgIdForCookieName(orgId)}.decisioning.`
+  );
+
+  const eventRegistry = createEventRegistry({ storage: storage.persistent });
+  let applyResponse = createApplyResponse();
+  const decisionProvider = createDecisionProvider();
+  const contextProvider = createContextProvider({ eventRegistry });
+
+  return {
+    lifecycle: {
+      onComponentsRegistered(tools) {
+        applyResponse = createApplyResponse(tools.lifecycle);
+      },
+      onBeforeEvent({
+        event,
+        renderDecisions,
+        decisionContext = {},
+        onResponse = noop
+      }) {
+        if (renderDecisions) {
+          onResponse(
```

Comment:
I can't think of a good use case at this point.  But doesn't mean we couldn't support it in the future.

Reply:
My point is that if you evaluate the rules before the request is sent, then personalization would have a chance to include the display notification inside the original event. Here you are running the evaluation in onResponse.

Reply:
Ah I see, so you are thinking it would be more efficient because we could include display events as part of the request rather than send them later.  That's fair.  The problem with that is that when a ruleset is satisfied, we only know the user has qualified for the item -- they have not yet seen it.  The display of the item is done by the personalization component which also sends the display event.  

I worry that doing as you suggest would give false positives, because some qualified items may not actually get displayed -- but display events would be captured.  Additionally, I like the separation of concerns.  Decisioning engine doesn't manage rendering items or determining if they have been displayed or not.